### PR TITLE
Configure Auto

### PR DESCRIPTION
**Pull request by tylerkrupicka**
## What is changing

I added the basic configuration for [auto](https://github.com/intuit/auto), with the following things configured:

- NPM releases (Release on Master merge, Canary versions)
- Released comments (when a PR is released, comment on the PR)
- All contributors (Automatically update all contributors when someone has a PR merged)

This PR will not work without some work from the person who runs the CircleCI builds, I'll try to document what you need to do below.

 - You'll need to set up an [initial tag and release](https://intuit.github.io/auto/docs/welcome/getting-started#make-latest-release). Looks like the current version is `1.0.2`.
- You'll need to set up [environment variables](https://intuit.github.io/auto/docs/welcome/getting-started#2-environment-variables). Auto will need a token to publish to NPM, and a token to write release notes to GitHub.
- You'll need to [add labels](https://intuit.github.io/auto/docs/welcome/getting-started#3-labels) for `patch`, `minor`, `major`, `documentation`, and `skip-release`, so it's easy to label PRs.

## What else might be impacted?

Mainly your build process. I re-worked it a bit to include auto in the pipeline and have more discrete steps. It's not going to work until we make some minor changes to the builds. Let me know if you need any help setting that up or testing!

## Which issue does this PR relate to?

Closes https://github.com/intuit/user-data-for-fraud-prevention/issues/24
Closes https://github.com/intuit/user-data-for-fraud-prevention/issues/11

## Checklist

[ ] Tests are written and maintain or improve code coverage
[ ] I've tested this in my application using `yarn link` (if applicable)
[ ] You consent to and are confident this change can be released with no regression

### CODE REVIEW

**Comment by tylerkrupicka** [comment]
I'm getting a circle `coveralls` error, but I didn't change that. Regardless this PR won't work without some Circle changes, so we can see if it works after updating the other things.

**Comment by reubenae** [comment]
Thanks @tylerkrupicka ! Really appreciate this. I've added the labels and will look into the CircleCI changes unless @skodamarthi gets there before me. Will check in with you if or when I get stuck.

**Comment by skodamarthi** [comment]
@tylerkrupicka Thanks for taking this up! I have done the first step of setting up initial tag and release for our latest version v1.0.3. About the 2nd step about environment variables, do we need to set this on local env file or can we try setting the environment variable on Circle CI and verify directly?

File: .circleci/config.yml
Comment (skodamarthi) [inline .circleci/config.yml:7]: Do we need to set this? Is this mandatory?

Comment (tylerkrupicka) [reply .circleci/config.yml:7]: Nope! I can remove it if you want.

**Comment by tylerkrupicka** [comment]
You'll need to set up the environment variables on CircleCI directly, the env file is usally just for local development and you don't commit it. You'll need to add the NPM token you use for publishing the package, and a GitHub token that has rights to the repo.

File: .circleci/config.yml
Comment (skodamarthi) [reply .circleci/config.yml:7]: Cool.. Please remove it for now.

**Comment by skodamarthi** [comment]
> You'll need to set up the environment variables on CircleCI directly, the env file is usally just for local development and you don't commit it. You'll need to add the NPM token you use for publishing the package, and a GitHub token that has rights to the repo.

Ok..So I need to create the following two environment variables to CircleCI right? GH_TOKEN and NPM_TOKEN
For NPM_TOKEN, I have generated a new one of type "Automation" to work with CI/CD workflows. Is that ok? Or do I need to generate "Publish" type?
Screen Shot 2020-10-07 at 00 08 31

**Comment by skodamarthi** [comment]
> I'm getting a circle `coveralls` error, but I didn't change that. Regardless this PR won't work without some Circle changes, so we can see if it works after updating the other things.

Not sure why coveralls failed with that error, but saw on some circleci discussion that the error should usually get fixed by itself if you submit another commit. https://discuss.circleci.com/t/admin-must-opt-in-to-3rd-part-orbs-but-i-already-have/35552. Let's see if it works after you make a new commit for removing timezone.

**Comment by tylerkrupicka** [comment]
@skodamarthi yeah an automation token should be what you want! I can make another commit to see if it works.

Thanks!

**Comment by skodamarthi** [comment]
> @skodamarthi yeah an automation token should be what you want! I can make another commit to see if it works.
> 
> Thanks!

Cool..I configured the environment variable on circleCI. And looks like CircleCI failed again. Not sure what is wrong in the config as it seems to be happening only with this PR.

File: .circleci/config.yml
Comment (skodamarthi) [inline .circleci/config.yml:7]: Could the location of orbs be a problem? Before your changes, It was placed at the beginning of the file. Not sure if it is an issue but just wondering..

**Comment by tylerkrupicka** [comment]
Reading the docs it seems like it may throw the error because I changed the config version to get newer syntax, which might make you re-allow third party orbs.

Can you see if https://circleci.com/docs/2.0/orbs-faq/#using-uncertified-orbs helps?

**Comment by skodamarthi** [comment]
> Reading the docs it seems like it may throw the error because I changed the config version to get newer syntax, which might make you re-allow third party orbs.
> 
> Can you see if https://circleci.com/docs/2.0/orbs-faq/#using-uncertified-orbs helps?

@tylerkrupicka I am not an admin for Intuit Organisation, so I am not able to change this setting. Is there any other way we can do this without having to change the setting?

**Comment by tylerkrupicka** [comment]
Well going back to your old config worked, although test coverage is throwing some sort of exception. Should I just leave the circle ci config unchanged, and then one of you can add the step for `auto shipit`. I'm not sure how circle handles changes to the build config in a PR.

**Comment by skodamarthi** [comment]
> Well going back to your old config worked, although test coverage is throwing some sort of exception. Should I just leave the circle ci config unchanged, and then one of you can add the step for `auto shipit`. I'm not sure how circle handles changes to the build config in a PR.

@tylerkrupicka Can the problem be because of adding 'auto shipit' to build job instead of release job in circle ci? Can you try moving it to release? I was just referring to circle ci configs in other intuit repos and have noticed that everywhere this step is done as part of release job. Eg: https://github.com/intuit/ReplayZero/blob/3faa5a68a44f3a8886ba359e19e7861035a2454f/.circleci/config.yml#L13

**Comment by tylerkrupicka** [comment]
Maybe related to these: https://github.com/lemurheavy/coveralls-public/issues/632
https://github.com/lemurheavy/coveralls-public/issues/1264

Did you change the token for coveralls when you changed the GH token? The weird thing is that would cause it to fail for all jobs on your Circle.

**Comment by skodamarthi** [comment]
@tylerkrupicka I tried now by regenerating the token for coveralls and adding the new token to circleCI environment variables. Still see the same build issue!

**Comment by skodamarthi** [comment]
@tylerkrupicka By the way, I am not sure why it is still building with  user-data-for-fraud-prevention@1.0.2..it should be for 1.0.3 as that is the latest version. 

Run tests and generate coverage report
4s
#!/bin/bash -eo pipefail
npm run test:coverage

> **user-data-for-fraud-prevention@1.0.2 test:coverage** /home/circleci/user-data-for-fraud-prevention
> npm test -- --coverage && cat ./coverage/lcov.info | coveralls

**Comment by reubenae** [comment]
What are the next steps on this PR? Can we merge it?

**Comment by tylerkrupicka** [comment]
@reubenae could probably use some help. Your coveralls integration is failing for this PR, but I didn't change anything related to it. It also appears [Josh has the same problem on his PR](https://github.com/intuit/user-data-for-fraud-prevention/pull/78). It's possible your builds fail when users make a PR from a fork. 

I think this is good to go otherwise.

**Comment by reubenae** [comment]
Looks like the fork thing is an issue: https://github.com/ethereumjs/ethereumjs-vm/issues/359

I'll check with Susmitha tomorrow but I suggest we merge this PR, if the master build passes we're good, if it fails I'll revert the merge. We'll probably need to disable coveralls if this is something it is going to do.

**Comment by skodamarthi** [comment]
@tylerkrupicka @reubenae I agree. I tried multiple approaches last week too by regenerating tokens, logging in/ logging out, removing and adding project again to coveralls etc; Nothing seemed to work for getting a successful build with these two PRs. We can merge these and I can disable coveralls from the ci build for now until we figure out the issue.

**Review by skodamarthi — APPROVED** [review]

**Comment by skodamarthi** [comment]
@tylerkrupicka After merging the PR, the master build was successful but the release failed with the below error. Any idea what could be wrong?
Screen Shot 2020-10-21 at 23 28 47

**Comment by tylerkrupicka** [comment]
@skodamarthi I think that was an oversight on my part, since auto is not being installed as a global name. change that line to `yarn run auto shipit`.

**Comment by reubenae** [comment]
@tylerkrupicka done https://github.com/intuit/user-data-for-fraud-prevention/pull/84 (cc @skodamarthi merge if you're happy)

**Comment by skodamarthi** [comment]
@tylerkrupicka We tried updating to "yarn run auto shipit" but it still failed with the below error:
Screen Shot 2020-10-22 at 10 55 09

**Comment by tylerkrupicka** [comment]
@skodamarthi Hmm yeah this is weird, I looked and actually we shouldn't have even had to make the other change, since scripts in package json already `yarn run`. I'm wondering if it's because in your CircleCI config there isn't an install step. I'm not super familiar with Circle, but maybe since you install in the build step, it's not available in the install step. Can you try making install a separate step that runs install, and make it a requirement for build and release?

I had it step up like that at one point but removed it when we were debugging coveralls.

Here's an example: https://github.com/intuit/ts-readme/blob/master/.circleci/config.yml

**Pull request merged**
